### PR TITLE
fix(nodejs): add support for parsing `workspaces` from `package.json` as an object

### LINKED ISSUE #6129: bug(nodejs): package.json parser fails on object notation for workspaces
Ref:
- JSON [schema](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/package.json#L777) `package.json`

### Discussed in https://github.com/aquasecurity/trivy/discussions/6114

<div type='discussions-op-text'>

<sup>Originally posted by **pjungermann** February 13, 2024</sup>
### Description

Currently, the `workspaces` field in package.json files is only allowed to be an array of strings. However, it has also an object version.

https://classic.yarnpkg.com/blog/2018/02/15/nohoist/

Example

```
  "workspaces": {
    "packages": [
      "packages/*",
      "plugins/*"
    ]
  },
```

Ref to an active open-source project using this notation:
https://github.com/backstage/backstage/blob/master/package.json#L52

Impact e.g. on Trivy scans:

```
WARN  Unable to parse "package.json" to remove dev dependencies: unable to parse .: parse error: JSON decode error: json: cannot unmarshal object into Go struct field packageJSON.workspaces of type []string
```

Relates-to: https://github.com/aquasecurity/go-dep-parser/issues/295

### Desired Behavior

Parsing package.json files does not fail when workspaces object notation is used.

### Actual Behavior

Parser fails and warning gets emitted:

```
WARN  Unable to parse "package.json" to remove dev dependencies: unable to parse .: parse error: JSON decode error: json: cannot unmarshal object into Go struct field packageJSON.workspaces of type []string
```

### Reproduction Steps

```bash
1. Close or create a project with workspaces object notation (e.g., https://github.com/backstage/backstage/blob/master/package.json#L52)
2. Run `trivy fs .`
```

### Target

Filesystem

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
-
```

### Operating System

MacOS Venture 13.6

### Version

```bash
2024-02-13T10:23:25.765+0100    INFO    Loaded trivy.yaml
Version: 0.47.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-02-13 06:11:52.656032521 +0000 UTC
  NextUpdate: 2024-02-13 12:11:52.656032281 +0000 UTC
  DownloadedAt: 2024-02-13 08:38:29.761065 +0000 UTC
Policy Bundle:
  Digest: sha256:1df8ade71efc830877ca3b1130f83e0c6368e3a45b0d4c0f0418955501644054
  DownloadedAt: 2023-11-24 02:43:02.401033 +0000 UTC
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)</div>

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add support for parsing `workspaces` from `package.json` as an object.
See more in #6129

## Related issues
- Close #6129

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/dependency/parser/nodejs/packagejson/parse.go [hunk 1]
```diff
@@ -73,3 +74,26 @@ func parseLicense(val interface{}) string {
 	}
 	return ""
 }
+
+// parseWorkspaces returns slice of workspaces
+func parseWorkspaces(val any) []string {
+	// Workspaces support 2 types - https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/package.json#L777
```
Comment (knqyf263) [inline pkg/dependency/parser/nodejs/packagejson/parse.go:80]:
```suggestion
	// Workspaces support 2 types - https://github.com/SchemaStore/schemastore/blob/d9516961f8a5b0e65a457808070147b5a866f60b/src/schemas/json/package.json#L777
```

Comment (knqyf263) [reply pkg/dependency/parser/nodejs/packagejson/parse.go:80]: The master branch will be updated.

Comment (DmitriyLewen) [reply pkg/dependency/parser/nodejs/packagejson/parse.go]: right. Thanks!

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
@DmitriyLewen It's conflicting now.

**Comment by DmitriyLewen** [comment]
Thanks! Resolved in https://github.com/aquasecurity/trivy/pull/6231/commits/5c1894cb6e4045d8928469ce45b22128797df5b5

**Pull request merged**